include/SFML/Config.hpp.in

```diff
@@ -35,10 +35,25 @@
 ////////////////////////////////////////////////////////////
 // Define the SFML version
 ////////////////////////////////////////////////////////////
-#define SFML_VERSION_MAJOR      3
-#define SFML_VERSION_MINOR      0
-#define SFML_VERSION_PATCH      0
-#define SFML_VERSION_IS_RELEASE false
+// clang-format off
+#define SFML_VERSION_MAJOR @SFML_VERSION_MAJOR@
+#define SFML_VERSION_MINOR @SFML_VERSION_MINOR@
+#define SFML_VERSION_PATCH @SFML_VERSION_PATCH@
+// clang-format on
+#cmakedefine01 SFML_VERSION_IS_RELEASE
+
+
+////////////////////////////////////////////////////////////
+// Define configuration constants
+////////////////////////////////////////////////////////////
+#cmakedefine SFML_OPENGL_ES
+#cmakedefine SFML_ANDROID_USE_SUSPEND_AWARE_CLOCK
+
+#if defined(SFML_OPENGL_ES)
+
+#define GL_GLEXT_PROTOTYPES
+
+#endif
```

Comment:
This now means that all users of SFML get these definitions as well as opposed the current implement which only applies the definitions to SFML itself. What's the rationale for that?

Reply:
It's sort of meta information about how SFML was built, similar to the release define. Now that you mention it, I should move `GL_GLEXT_PROTOTYPES` to where it's actually used, so that doesn't leak.

Reply:
What's the reason a user would need access to these symbols?

Reply:
I guess not much, I decided to put it here just for simplicity. I will just use a separate, private header.

Reply:
I'd rather keep the current solution than introduce even more generated headers

Reply:
Actually, both `SFML_OPENGL_ES` and `SFML_ANDROID_USE_SUSPEND_AWARE_CLOCK` affect the public API (and ABI?) so they should be in the config header. `GL_GLEXT_PROTOTYPES` can be moved to `src/SFML/CMakeLists.txt` as a scoped compile definition.

For the clock one, `std::chrono::high_resolution_clock` _may be_ an alias for `std::chrono::steady_clock`. ([doc](https://en.cppreference.com/w/cpp/chrono/high_resolution_clock))  
For the OpenGL one, different headers are included based on the definition. ([source](https://github.com/SFML/SFML/blob/a71d60a0c91881f3c72d1bcda232f5a42fb91d31/include/SFML/OpenGL.hpp#L54-L59))

Reply:
Are you saying the current state of `master` is buggy since as far as I can tell `SFML_OPENGL_ES` and `GL_GLEXT_PROTOTYPES` are not usage requirements passed onto downstream targets?

Reply:
`GL_GLEXT_PROTOTYPES` is only used as a build requirement, but `SFML_OPENGL_ES` appears in the public interface and includes different transitive headers if set.

Reply:
I'd love to continue this discussion in a separate PR if you don't mind reducing the scope of this one. I like the core change to Config.hpp.in and want to get those merged first.